### PR TITLE
[Merged by Bors] - Review

### PR DESCRIPTION
**Pull request by zkf331**

### CODE REVIEW

**Review by priscilla-pan — CHANGES_REQUESTED** [review]
please  delete .idea directory

File: src/object_detection/utils/models.py [hunk 1]
```diff
@@ -0,0 +1,477 @@
...
+        if isinstance(cfg, str):
+            self.module_dicts = parse_model_cfg(cfg)
+        elif isinstance(cfg, list):
+            self.module_dicts = cfg
+
+        self.module_list, self.routs, self.hyperparams = create_modules(self.module_dicts, img_size, arc)
+        self.yolo_layers = _get_yolo_layers(self)
+
+        # Darknet Header https://github.com/AlexeyAB/darknet/issues/2914#issuecomment-496675346
+        self.version = np.array([0, 2, 5], dtype=np.int32)  # (int32) version info: majork, minor, revision
```
Comment (ChengcanWang-zte) [inline src/object_detection/utils/models.py]: may be major, minor, revision

File: src/object_detection/utils/models.py [hunk 2]
```diff
@@ -0,0 +1,477 @@
...
+        return infer_output.view(batch_size, -1, 5 + self.num_classes), pred
+
+
+def _get_yolo_layers(models):
+    """the index of YOLO output layer"""
+
+    return [i for i, x in enumerate(models.module_dicts) if x['type'] == 'yolo']
+
+
+def create_grids(self, img_size=608, num_grid=(13, 13), devices='cpu', torch_type=torch.float32):
```
Comment (ChengcanWang-zte) [inline src/object_detection/utils/models.py]: Why not write this as a class method?

File: src/object_detection/utils/models.py [hunk 3]
```diff
@@ -0,0 +1,477 @@
...
+
+    # build wh gains
+    self.anchor_vec = self.anchors.to(devices) / self.stride
+    self.anchor_wh = self.anchor_vec.view(1, self.num_anchors, 1, 1, 2).to(devices).type(torch_type)
+    self.num_grid = torch.Tensor(num_grid).to(devices)
+    self.num_x = num_x
+    self.num_y = num_y
+
+
+def load_darknet_weights(self, weights, cutoffs=-1):
```
Comment (ChengcanWang-zte) [inline src/object_detection/utils/models.py]: Why not write this as a class method?

File: src/object_detection/utils/dataset.py [hunk 4]
```diff
@@ -0,0 +1,591 @@
...
+            label4[:, 2] = height * (label[:, 2] - label[:, 4] / 2) + padh
+            label4[:, 3] = width * (label[:, 1] + label[:, 3] / 2) + padw
+            label4[:, 4] = height * (label[:, 2] + label[:, 4] / 2) + padh
+        labels4.append(label4)
+
+    # Concat/clip labels
+    if labels4:
+        labels4 = np.concatenate(labels4, 0)
+        np.clip(labels4[:, 1:], 0, 2 * img_size, out=labels4[:, 1:])  # use with random_perspective
+        # img4, labels4 = replicate(img4, labels4)  # replicate
```
Comment (ChengcanWang-zte) [inline src/object_detection/utils/dataset.py]: Remove useless code

File: .idea/deployment.xml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  <component name="PublishConfigData" autoUpload="Always" serverName="root@10.42.10.123:30123">
+    <serverData>
+      <paths name="root@10.42.10.123:30123">
+        <serverdata>
+          <mappings>
+            <mapping deploy="/root/work/Adlik_object_detection" local="$PROJECT_DIR$" web="/" />
+          </mappings>
+        </serverdata>
+      </paths>
+      <paths name="root@10.42.10.123:30123 (1)">
```
Comment (shiyingjin) [inline .idea/deployment.xml]: Commit too many pirvate informations

**Review by shiyingjin — CHANGES_REQUESTED** [review]

File: src/object_detection/utils/prune_utils.py [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+
+    :param output_student: the output of student model
+    :param output_teacher:  the output of teacher model
+    :param num_classes:  classes number
+    :param batch_size:  batch size
+    :return: loss_st * Lambda_ST
+    """
+
+    lambda_st = 0.01
+    criterion_st = torch.nn.KLDivLoss(reduction='sum')
```
Comment (ChengcanWang-zte) [inline src/object_detection/utils/prune_utils.py]: reduction='batchmean'  may be better than sum, 'sum' will make difference effect with batchsize.

File: src/object_detection/train.py [hunk 7]
```diff
@@ -0,0 +1,394 @@
...
+                if mixed_precision:
+                    with torch.no_grad():
+                        output_teacher = teacher_model(imgs)
+                else:
+                    _, output_teacher = teacher_model(imgs)
+                soft_target = distillation_loss(pred, output_teacher, model.num_classes, imgs.size(0))
+                loss += soft_target
+
+            # Scale loss by nominal batch_size of 64
+            loss *= batch_size / 64
```
Comment (ChengcanWang-zte) [inline src/object_detection/train.py:309]: Loss with mean, distill loss with sum. May be not match

File: .pylintrc [hunk 8]
```diff
@@ -1,10 +1,25 @@
 [MASTER]
 jobs = 0
+extension-pkg-whitelist=numpy,torch
+init-hook="from pylint.config import find_pylintrc; import os, sys; sys.path.append(os.path.dirname(find_pylintrc()))"
 
 [MESSAGES CONTROL]
 disable = missing-class-docstring,
           missing-function-docstring,
-          missing-module-docstring
+          missing-module-docstring,
```
Comment (shiyingjin) [inline .pylintrc]: Too many disable controls

**Review by shiyingjin — CHANGES_REQUESTED** [review]

File: .pylintrc (hunk 8 above)
Comment (zkf331) [reply .pylintrc]: Most disrables have been modified. The rest of the disables takes time to make major changes to the code，The next version iteration will modify it.
